lib/cookbook.rb

```diff
@@ -0,0 +1,17 @@
+class CookBook
+  attr_reader:recipes
+  def initialize
+    @recipes =[]
+  end
+
+  def add_recipe(recipe)
+    @recipes << recipe
+  end
+
+  def ingredients
+    ingredients = []
+    @recipes.each do |ingredient|
+    ingredients <<  ingredient.name[ingredient]  
```

Comment:
Not sure what that bracket notation is accomplishing. This would have worked without it, then returning the ingredients array on the last line.